luaui/Widgets/gui_chat.lua

```diff
@@ -1888,7 +1901,14 @@ function widget:KeyPress(key)
 					if ssub(inputText, 1, 1) == '/' then
 						Spring.SendCommands(ssub(inputText, 2))
 					else
-						Spring.SendCommands("say "..inputMode..inputText)
+						local badWord = findBadWords(inputText)
+						if badWord ~= nil and inputText ~= lastMessage then
+							addChat(Spring.GetGameFrame(), LineTypes.System, "\255\255\000\000Moderation:",
+							'Words such as "' .. badWord .. '" can be against the Code of Conduct if they are being used to abuse other players. Please take a moment to reevaluate your intentions. If your message was blocked in error, then you can send it again to bypass the filter.', true)
```

Comment:
This message needs to support I18N. We should not be hardcoding any display text.